File: 2656-maximum-sum-with-exactly-k-elements/2656-maximum-sum-with-exactly-k-elements.rs
use std::collections::BinaryHeap;

impl Solution {
    pub fn maximize_sum(nums: Vec<i32>, k: i32) -> i32 {
        let mut pq = BinaryHeap::new();
        let mut res = 0;
        for num in nums {
            pq.push(num);
        }
        
        for _ in 0..k {
            let top_elem = pq.pop().unwrap();
            res += top_elem;
            
            let next_elem = top_elem + 1;
            pq.push(next_elem)
        }
        
        
        res
    }
}

File: 2656-maximum-sum-with-exactly-k-elements/2656-maximum-sum-with-exactly-k-elements.py
import heapq


class Solution:
    def maximizeSum(self, nums: List[int], k: int) -> int:
        pq = []
        res = 0

        for num in nums:
            heapq.heappush(pq, -num)

        while pq and k > 0:
            top_element = -heapq.heappop(pq)
            res += top_element

            next_element = top_element + 1
            heapq.heappush(pq, -next_element)

            k -= 1

        return res
